### PR TITLE
Group docs fixes

### LINKED ISSUE #23: Group has two clear methods in API, removeSprites not listed
The API lists two clear methods. The code has @method clear on the removeSprites method. clear API
clear documentation

### PR DESCRIPTION
**Pull request by toolness**
This fixes #23 and #64. It doesn't regenerate the documentation, though, as I didn't want to spam the commit history... :disappointed:

### CODE REVIEW

File: lib/p5.play.js [hunk 1]
```diff
@@ -2712,7 +2712,7 @@ function Group() {
   * </code>
   *
   * @method overlap
-  * @param {Object} target Group to check against the current one
+  * @param {Object} target Group or Sprite to check against the current one
```
Comment (islemaster) [inline lib/p5.play.js]: You could be more specific in the parameter type here (pretty sure; YUIDoc reference for [@type](http://yui.github.io/yuidoc/syntax/index.html#type), [@param](http://yui.github.io/yuidoc/syntax/index.html#param)):

```
* @param {Group|Sprite} Group or Sprite to check against the current one
```

**Comment by islemaster** [comment]
Once comment on documenting types, otherwise looks good to me!

**Comment by islemaster** [comment]
Want to fix #63 while you're at it here?

File: lib/p5.play.js (hunk 1 above)
Comment (toolness) [inline lib/p5.play.js]: Oh, I had considered that, but I figured there were other places where `{Object}` was used in place of something more specific, so I just changed the wording... Do you mind if we file a separate PR for that so I can change all the `{Object}` types in one go?

**Comment by toolness** [comment]
> Want to fix #63 while you're at it here?

Actually do you mind if I do that in a separate PR? I wanted to keep this one only focused on the docs for `Group`... maybe it's a bit pedantic but I guess I prefer a bunch of tiny focused PRs to bigger ones. Granted, I guess this one isn't particularly focused to begin with. :)

**Comment by toolness** [comment]
Uh, also, it turns out that figuring out exactly *how* we want to fix #63 is actually a bit non-trivial, so I've filed https://github.com/molleindustria/p5.play/pull/67 separately. :grin:

**Comment by islemaster** [comment]
If you want to break stuff out, I'm all for it.  This LGTM!

**Comment by toolness** [comment]
Cool thanks!

**Pull request merged**